# Урок 5. Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

In [2]:
import numpy as np

### Задание 1

Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

In [3]:
std = 16
dov_int = 0.95
a = 0.05
M = 80
n =256

In [4]:
M+(1.96 * 16 / np.sqrt(n))

81.96

In [5]:
M-(1.96 * 16 / np.sqrt(n))

78.04

Доверительный интервал для оценки математического ожидания с надежностью 0.95 [78.04;81.96]

### Задание 2

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

Не смещенная D(X):

In [6]:
M_ = np.mean([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
M_

6.590000000000001

In [7]:
std_ = np.std([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1], ddof=1)
std_

0.4508017549014448

По таблице Стьюдента t = 2.262

In [11]:
t = 2.262

In [12]:
M_ + (t * std_ / np.sqrt(10))

6.912461744087575

In [13]:
M_ - (t * std_ / np.sqrt(10))

6.267538255912426

Доверительный интервал [6.27;6.91]

### Задание 3

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.

In [27]:
H_0 = 17
H_1 = 17.5
a = 0.05
n = 100
Z_t = 1.645
d = 4
std = np.sqrt(d)

In [28]:
Z_n = (H_1 - H_0)/(std/np.sqrt(n))
Z_n

2.5

Ответ: $Z_н=2.5 > Z_т=1.65.$ H_1 верна при одностороннем критерии $\alpha=5$%

### Задание 4

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [16]:
M_2 = np.mean([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
M_2

198.5

In [17]:
std_2 = np.std([202, 203, 199, 197, 195, 201, 200, 204, 194, 190],ddof=1)
std_2

4.453463071962462

$\alpha$=1%, $t, где _{\alpha/2}$=0.05%. По таблице Стьюдента t = 3.250

In [18]:
t_2 = 3.250

In [19]:
M_2 + (t_2 * std_2 / np.sqrt(10))

203.0770032044268

In [20]:
M_2 - (t_2 * std_2 / np.sqrt(10))

193.9229967955732

Если учитывать, что доверительная вероятность равна 99%, то утверждение продавца верно, поскольку 200 попадает в доверительный интервал [194;203]. Сначала так понял задание. Можно так утверждать вообще?

In [30]:
tn = (M_2 - 200)/(std_2/np.sqrt(10))
tn

-1.0651074037450896

$|tn|<t_2$, можно утверждать, что гипотеза верна